File: 024. Generate Document/GenerateDocument.java
import java.util.*;

class GenerateDocument {

    public boolean generateDocument(String characters, String document) {
        HashMap<Character, Integer> characterCounts = new HashMap<Character, Integer>();

        for (int i = 0; i < characters.length(); i++) {
            char character = characters.charAt(i);
            characterCounts.put(character, characterCounts.getOrDefault(character, 0) + 1);
        }

        for (int i = 0; i < document.length(); i++) {
            char character = document.charAt(i);
            if (!characterCounts.containsKey(character) || characterCounts.get(character) == 0) {
                return false;
            }
            characterCounts.put(character, characterCounts.get(character) - 1);
        }

        return true;
    }
}

File: 024. Generate Document/GenerateDocument.kt
package com.algoexpert.program

fun generateDocument(characters: String, document: String): Boolean {
    val characterCounts = mutableMapOf<Char, Int>()

    for (char in characters) {
        if (!(char in characterCounts)) characterCounts[char] = 0
        characterCounts[char] = characterCounts[char]!! + 1
    }

    for (char in document) {
        if (!(char in characterCounts) || characterCounts[char] == 0) return false
        characterCounts[char] = characterCounts[char]!! - 1
    }

    return true
}
